app/main.py

```diff
@@ -2,0 +1,37 @@
+from __future__ import annotations
+
+
+class Animal:
+
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden : bool = False
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    @classmethod
+    def health_drop(cls) -> None:
+        cls.alive = [animal for animal in cls.alive if animal.health > 0]
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(herbivore : Animal) -> None:
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                Animal.health_drop()
```

Comment:
The `health_drop` method is called only when a `Carnivore` bites a `Herbivore`. It should be called whenever an animal's health changes to ensure the `alive` list is always up-to-date.